File: CosechasApp/app/src/main/java/com/techtraining/cosechasapp/db/Camion.java
package com.techtraining.cosechasapp.db;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
@Entity
public class Camion {
    @PrimaryKey
    public int id;

    @ColumnInfo(name = "tipo_camion")
    public String tipoCamion;

    @ColumnInfo(name = "ancho")
    public double ancho;

    @ColumnInfo(name = "placa")
    public String placa;

    @ColumnInfo(name = "alto")
    public double alto;

    @ColumnInfo(name = "empresa_id")
    public int empresaId;

    @ColumnInfo(name = "camionero")
    public String camionero;

    @ColumnInfo(name = "identificacion_camionero")
    public String identificacionCamionero;

    @ColumnInfo(name = "estado")
    public String estado;
    @Override
    public String toString() {
        return this.placa;
    }
}

File: ProyectoBaseWebAppSaul/app/Camion.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Camion extends Model
{
    protected $table = 'camiones';
    protected $fillable = [
        'tipo_camion', 'ancho', 'placa', 'alto', 'empresa_id', 'camionero', 'identificacion_camionero', 'estado', 'creador_id', 'modificador_id',
        'codigo_vendor', 'filas'
    ];
    protected $appends = [
        'descripcion_tipo_camion'
    ];
    public function getDescripcionTipoCamionAttribute()
    {
        switch ($this->tipo_camion) {
            case 'B':
                return 'Bananero';
            case 'T':
                return 'Trailer';
            default:
                return 'No definido';
        }
    }
    public function scopeActive($query)
    {
        return $query->where('estado', 'A');
    }
    public function creador()
    {
        return $this->belongsTo('App\User');
    }
    public function modificador()
    {
        return $this->belongsTo('App\User');
    }
    public function scopeCurrent($query, $empresaId)
    {
        return $query->where('empresa_id', $empresaId);
    }
}
